### PR TITLE
Electron-338 (Replace html tags from message content)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Replace HTML tag `<>` from message content to prevent HTML injection [ELECTRON-338](https://perzoinc.atlassian.net/browse/ELECTRON-338)

## Approach
How does this change address the problem?
#### Problem with the code:
- Messages with HTML content were not being filtered and was able to inject arbitrary HTML code into notifications
#### Fix:
- Replacing HTML tags `<>` with the entity names

### Before
screen shot 2018-03-12 at 4 58 15 pm

### After
screen shot 2018-03-12 at 4 58 55 pm

## Spectron test results
[Electron-338 — Spectron.pdf](https://github.com/symphonyoss/SymphonyElectron/files/1803034/Electron-338.Spectron.pdf)

## Unit test results
[Electron-338 — Unit Tests.pdf](https://github.com/symphonyoss/SymphonyElectron/files/1803036/Electron-338.Unit.Tests.pdf)

## Open Questions if any and Todos
- [X] Unit-Tests (Not Applicable)
- [X] Documentation
- [ ] Automation-Tests
When solved, check the box and explain the answer.

@VikasShashidhar & @VishwasShashidhar Please review. Thanks 😄

### CODE REVIEW

File: js/notify/notifyImpl.js [hunk 1]
```diff
@@ -241,4 +246,46 @@ function Queue(emitter) {
...
+    return string.replace(/[<>]/g, replaceTag);
+}
+
+/**
+ * Replace strong HTML tags from the string
+ *
+ * @param string
+ * @return {void | string | *}
+ */
+// TODO(KiranNiranjan): Can be removed after 2-3 sprints
```
Comment (VikasShashidhar) [inline js/notify/notifyImpl.js]: Remove this please and instead create a ticket on JIRA.

File: js/notify/notifyImpl.js [hunk 2]
```diff
@@ -241,4 +246,46 @@ function Queue(emitter) {
...
 }
 
+/**
+ * Replace HTML tags <> from messages test to prevent
+ * HTML injection
+ *
+ * @param string
+ * @return {void | string | *}
+ */
+function replaceHTMLTags(string) {
```
Comment (VikasShashidhar) [inline js/notify/notifyImpl.js]: Could rename the param to `tag` perhaps?

File: js/notify/notifyImpl.js [hunk 3]
```diff
@@ -241,4 +246,46 @@ function Queue(emitter) {
...
+}
+
+/**
+ * Replace strong HTML tags from the string
+ *
+ * @param string
+ * @return {void | string | *}
+ */
+// TODO(KiranNiranjan): Can be removed after 2-3 sprints
+function replaceStrongTag(string) {
```
Comment (VikasShashidhar) [inline js/notify/notifyImpl.js]: Same here.

File: js/notify/notifyImpl.js (hunk 2 above)
Comment (VishwasShashidhar) [reply js/notify/notifyImpl.js]: @KiranNiranjan Yes, please rename the parameter.

File: js/notify/notifyImpl.js (hunk 3 above)
Comment (VishwasShashidhar) [reply js/notify/notifyImpl.js]: @VikasShashidhar No, we need this to be called for backward compatibility reasons.

File: js/notify/notifyImpl.js (hunk 1 above)
Comment (VishwasShashidhar) [reply js/notify/notifyImpl.js]: @KiranNiranjan Yes, please raise a JIRA ticket.

File: js/notify/notifyImpl.js (hunk 3 above)
Comment (VikasShashidhar) [reply js/notify/notifyImpl.js]: I was actually talking about the ToDo comment above that.

File: js/notify/notifyImpl.js (hunk 2 above)
Comment (KiranNiranjan) [reply js/notify/notifyImpl.js]: @VikasShashidhar & @VishwasShashidhar But its a `String`, It may or may not contain HTML tags

Comment (VikasShashidhar) [reply js/notify/notifyImpl.js]: I see. Is there something else we can name it to? The line `string !== 'string'` is a little confusing IMO.

Comment (KiranNiranjan) [reply js/notify/notifyImpl.js]: Shall I rename it to `message`?

Comment (VishwasShashidhar) [reply js/notify/notifyImpl.js]: yes please 🙂

**Comment by VishwasShashidhar** [comment]
lgtm 👍

**Pull request merged**